src/pkcs15init/pkcs15-isoApplet.c

```diff
@@ -548,6 +548,7 @@ isoApplet_generate_key_ec(const sc_pkcs15_prkey_info_t *key_info, sc_card_t *car
 	memcpy(pubkey->u.ec.ecpointQ.value, args.pubkey.ec.ecPointQ.value, args.pubkey.ec.ecPointQ.len);
 
 	/* The OID is also written to the pubkey->u.ec.params */
+	free(pubkey->u.ec.params.der.value);
```

Comment:
It would help if this would be a bit more described. In the past, the EC params were sometimes very wildly assigned to various structures without much care. I hope this has improved a bit over last year with the fuzzing and with this PR.

For this case, the `pubkey` comes from `sc_pkcs15init_generate_key()` function, where it is memset to zeroes and then the EC params are copied to that structure from the keygen params using `sc_copy_ec_params()` so it should not really be anything dangerous (unless somebody starts assigning here some dangerous values on the way).

Reply:
Should I drop some of the patches/changes from the PR to get it merged (without these changes, the CI will most likely fail)? Or do you mean that we should just create a follow-up issue to review how the ec->params are handled in various cases?